newIDE/app/src/InstancesEditor/ViewPosition.js

```diff
@@ -103,9 +104,17 @@ export default class ViewPosition {
     this.viewY = y;
   }
 
-  scrollToInstance(instance: gdInitialInstance) {
-    this.viewX = instance.getX();
-    this.viewY = instance.getY();
+  /**
+   * Moves view to the rectangle center and returns the ideal zoom
+   * factor to fit to the rectangle.
+   */
+  fitToRectangle(rectangle: Rectangle) {
+    this.viewX = rectangle.centerX();
+    this.viewY = rectangle.centerY();
+    const idealZoomOnX = this._width / rectangle.width();
+    const idealZoomOnY = this._height / rectangle.height();
+
+    return Math.min(idealZoomOnX, idealZoomOnY) - 0.05; //Add margin so that the object don't feel cut
```

Comment:
Zoom are ratios they can't be subtracted. Maybe `* 0.95` ?

Reply:
Good idea